src/parser/mod.rs

```diff
@@ -3742,24 +3742,23 @@ impl<'a> Parser<'a> {
             });
         }
         self.expect_token(&Token::LParen)?;
-        let in_op = if self.parse_keyword(Keyword::SELECT) || self.parse_keyword(Keyword::WITH) {
-            self.prev_token();
-            Expr::InSubquery {
+        let in_op = match self.try_parse(|p| p.parse_query_body(p.dialect.prec_unknown())) {
```

Comment:
The previous approach peeked for a `SELECT` or `WITH`, but we cannot rely on that because there may be more than one parenthesis, as in `IN ((SELECT ...) UNION ())`.

The new approach tries to parse a subquery first, then an in list second.

Reply:
Yeah this makes sense!